src/scores/utils.py

```diff
@@ -39,13 +40,139 @@
 """
 
 
-class DimensionError(Exception):
+class DimensionError(ValueError):
```

Comment:
Why should this inherit from ValueError? I remember trying to think of the right thing to do about this class, I'd be interested in your thinking.

Reply:
It just adds an intermediate layer. E.g. if you wanted to catch all `ValueError`s but not all `Exception`s.